dashboard/app/helpers/levels_helper.rb

```diff
@@ -345,13 +345,18 @@ def match_t(text)
 
   def level_title
     if @script_level
-      if @script_level.script.legacy_curriculum?
-        "#{data_t('game.name', @game.name)} #{'#' + @script_level.position.to_s unless @script_level.position == 1}"
+      script = data_t_suffix('script.name', @script_level.script.name, 'title')
+      stage = @script_level.name
+      position = @script_level.position
+      if @script_level.script.stages.many?
+        "#{script}: #{stage} ##{position}"
```

Comment:
@tanyaparker?  If not I'll address post-merge, since most puzzles already use this pattern.  Previous code: https://github.com/code-dot-org/code-dot-org/pull/1495/files#diff-1f68e16ab551449bdbd0fb7f01004f68L351